src/librustc_typeck/check/op.rs

```diff
@@ -562,7 +563,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                    on the left and may require reallocation. This \
                    requires ownership of the string on the left";
 
-        let is_std_string = |ty| &format!("{:?}", ty) == "std::string::String";
+        let is_std_string =
+            |ty| with_crate_prefix(|| &format!("{:?}", ty) == "std::string::String");
```

Comment:
Should be diagnostic item.

Reply:
Indeed, like it was done for traits - I'll try to fix it in a separate PR.

Reply:
Pushed as #74471